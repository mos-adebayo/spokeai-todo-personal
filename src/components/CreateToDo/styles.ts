import styled from "styled-components";

export const FormWrapper = styled.div`
  width: 50%;
  margin: 20px auto;
  
  .form-control:focus {
    box-shadow: none;
  }
`;
