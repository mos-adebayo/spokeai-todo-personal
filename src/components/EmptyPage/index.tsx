import React from 'react';

const Index: React.FC = () => {
    return (
        <div>
            Your notes will here!
        </div>
    );
};

export default Index;
