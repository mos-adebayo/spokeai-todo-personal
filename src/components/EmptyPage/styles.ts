import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  margin: 20px auto;
  text-align: center;

  @media screen and (min-width: 768px) {
    width: 80%;
  }

  @media screen and (min-width: 992px) {
    width: 60%;
  }

  @media screen and (min-width: 1200px) {
    width: 50%;
  }
`;

export const LinkText = styled.span`
  cursor: pointer;
  text-decoration: underline;
`;
