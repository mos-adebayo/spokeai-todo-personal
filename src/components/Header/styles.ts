import styled from "styled-components";
import { gradientPrimaryBackground } from "../../util/stylesheet";

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 45px 20px;
  height: 100px;
  margin-bottom: 25px;
  font-family: "Mochiy Pop One", sans-serif;

  ${gradientPrimaryBackground};
`;
