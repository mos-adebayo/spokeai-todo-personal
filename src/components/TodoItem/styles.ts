import styled from "styled-components";

export const Wrapper = styled.div`
  background: #fff;
  border-radius: 12px;
  box-shadow: rgb(166 171 189 / 20%) 0 10px 20px;
  padding: 15px 30px;
  margin-bottom: 15px;
`;

export const Title = styled.h4`
color: #49486c;
`;

export const Text = styled.p`
color: #49486c
`;
