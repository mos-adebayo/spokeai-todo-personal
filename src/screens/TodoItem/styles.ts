import styled from "styled-components";

export const Wrapper = styled.div`
  width: 100%;
  margin: 0 auto;
  background: #fff;
  height: calc(100vh - 145px);

  @media screen and (min-width: 576px) {
    width: 95%;
  }

  @media screen and (min-width: 768px) {
    width: 70%;
  }

  @media screen and (min-width: 1200px) {
    width: 50%;
  }
`;
