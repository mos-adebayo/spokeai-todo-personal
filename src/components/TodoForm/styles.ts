import styled, { css } from "styled-components";
import { Button } from "react-bootstrap";
import { CheckBox } from "../TodoItem/styles";

export const AddButton = styled(Button)`
  &:focus {
    box-shadow: none;
  }

  background: linear-gradient(
    77.07deg,
    rgb(69, 183, 204) 0%,
    rgb(75, 207, 187) 100%
  );
  outline: none;
  border: none;
`;

export const AddButtonWrapper = styled.div`
  position: absolute;
  right: 50px;
  bottom: 50px;

  display: flex;
  justify-content: center;
  align-items: center;

  color: #ffffff;
  cursor: pointer;

  width: 100px;
  height: 100px;
  font-size: 80px;
  border-radius: 50%;
  background: linear-gradient(
    77.07deg,
    rgb(69, 183, 204) 0%,
    rgb(75, 207, 187) 100%
  );
  box-shadow: rgb(52 58 64 / 40%) 3px 1px 13px 0;

  @media screen and (max-width: 767px) {
    width: 80px;
    height: 80px;
    font-size: 60px;
  }
`;

export const FormWrapper = styled.div`
  margin: 20px 0;

  .form-control:focus {
    box-shadow: none;
  }
`;

export const ItemWrapper = styled.div`
  border-bottom: 1px solid #dfdfdf;
  margin: 0 -16px;
  padding: 10px 16px 8px;

  ${CheckBox} {
    margin-right: 5px;
  }

  &:first-of-type {
    border-top: 1px solid #dfdfdf;
  }
`;

const editItemWrapper = css`
  ${ItemWrapper} {
    margin: 0;
  }
`;

export const ItemsWrapper = styled.div<{ editMode?: boolean }>`
  margin-top: 15px;

  .form-control {
    height: 1.5rem;
    border: none;
  }

  ${({ editMode }) => editMode && editItemWrapper};
`;
