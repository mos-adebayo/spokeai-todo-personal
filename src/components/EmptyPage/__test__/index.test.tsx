import React from "react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { mount, ReactWrapper } from "enzyme";
import EmptyPage from "../index";
import { LinkText } from "../styles";

const mockStore = configureStore();

describe("<EmptyPage />", () => {
  let wrapper: ReactWrapper;
  beforeEach(() => {
    wrapper = mount(
      <Provider store={mockStore()}>
        <EmptyPage />
      </Provider>
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should contain the link to trigger creation", () => {
    const link = wrapper.find(LinkText);
    expect(link.length).toBe(1);
  });
});
