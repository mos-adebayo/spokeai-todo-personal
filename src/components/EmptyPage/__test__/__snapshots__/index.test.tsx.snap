// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EmptyPage /> should match snapshot 1`] = `
.c0 {
  width: 100%;
  margin: 20px auto;
  text-align: center;
}

.c1 {
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media screen and (min-width:768px) {
  .c0 {
    width: 80%;
  }
}

@media screen and (min-width:992px) {
  .c0 {
    width: 60%;
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    width: 50%;
  }
}

<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <EmptyPage>
    <styled.div>
      <div
        className="c0"
      >
        <Alert
          closeLabel="Close alert"
          show={true}
          transition={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "appear": false,
                "in": false,
                "mountOnEnter": false,
                "timeout": 300,
                "unmountOnExit": false,
              },
              "render": [Function],
            }
          }
          variant="info"
        >
          <Fade
            appear={false}
            in={true}
            mountOnEnter={false}
            timeout={300}
            unmountOnExit={true}
          >
            <ForwardRef
              addEndListener={[Function]}
              appear={false}
              childRef={null}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              timeout={300}
              unmountOnExit={true}
            >
              <Transition
                addEndListener={[Function]}
                appear={false}
                enter={true}
                exit={true}
                in={true}
                mountOnEnter={false}
                nodeRef={
                  Object {
                    "current": .c0 {
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

@media screen and (min-width:768px) {

}

@media screen and (min-width:992px) {

}

@media screen and (min-width:1200px) {

}

<div
                      class="fade alert alert-info show"
                      role="alert"
                    >
                      <p>
                        Your Todos will appear here!
                        <span
                          class="c0"
                        >
                          Get started
                        </span>
                      </p>
                    </div>,
                  }
                }
                onEnter={[Function]}
                onEntered={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                onExiting={[Function]}
                timeout={300}
                unmountOnExit={true}
              >
                <div
                  className="fade alert alert-info show"
                  role="alert"
                >
                  <p>
                    Your Todos will appear here!
                    <styled.span
                      onClick={[Function]}
                    >
                      <span
                        className="c1"
                        onClick={[Function]}
                      >
                        Get started
                      </span>
                    </styled.span>
                  </p>
                </div>
              </Transition>
            </ForwardRef>
          </Fade>
        </Alert>
      </div>
    </styled.div>
  </EmptyPage>
</Provider>
`;
